src/promptflow/promptflow/batch/_csharp_executor_proxy.py

```diff
@@ -46,9 +48,34 @@ def chat_output_name(self) -> Optional[str]:
         return self._chat_output_name
 
     @classmethod
-    def generate_metadata(cls, flow_file: Path, assembly_folder: Path):
-        """Generate metadata for the flow and save them to files under .promptflow folder.
-        including flow.json and flow.tools.json.
+    def generate_flow_metadata(
+        cls,
+        flow_file: Path,
+        working_dir: Path,
+        dump: bool = True,
+        timeout: int = FLOW_META_JSON_GEN_TIMEOUT,
+        load_in_subprocess: bool = True,
+    ) -> Dict[str, Any]:
+        # TODO: timeout & dump doesn't take effect for now
+        # TODO: provide a way to skip dumping and directly read from flow.json
+        cls._dump_metadata(
+            flow_file=flow_file,
+            working_dir=working_dir,
+        )
+
+        from promptflow import load_flow
+        from promptflow._sdk.entities._eager_flow import EagerFlow
+
+        flow = load_flow(flow_file)
+        if isinstance(flow, EagerFlow):
+            return json.load((working_dir / PROMPT_FLOW_DIR_NAME / FLOW_META_JSON).open())
```

Comment:
Do we need to check whether file exists here? provide a better exeption?

Reply:
updated and merge the logic with flow.tools.json check